src/overlays/actors/ovl_En_Syateki_Man/z_en_syateki_man.h

```diff
@@ -4,49 +4,70 @@
+    /* 4 */ SG_GAME_STATE_ONE_MORE_GAME,    // The player failed to get a new high score (Town) and/or perfect score (Swamp and Town).
+    /* 5 */ SG_GAME_STATE_GIVING_BONUS,     // The player gets bonus points at the end of the Swamp game if they get a perfect score.
+    /* 6 */ SG_GAME_STATE_ENDED,            // The player got a new high score and/or perfect score (Town), or the game is over (Swamp).
+    /* 7 */ SG_GAME_STATE_MOVING_PLAYER,    // The player is automatically moving towards the spot to play the game.
+} ShootingGalleryGameState;
+
+typedef enum {
+    /*  0 */ SG_OCTO_STATE_SPAWNING,
+    /*  1 */ SG_OCTO_STATE_SPAWNED,
+    /* 70 */ SG_OCTO_STATE_INITIAL = 70,
+    /* 80 */ SG_OCTO_STATE_HIDING = 80,
+} ShootingGalleryOctorokState;
 
-typedef struct {
-    /* 0x00 */ s16 index;
-    /* 0x04 */ f32 x;
-    /* 0x08 */ f32 y;
-    /* 0x0C */ f32 z;
-    /* 0x10 */ s32 variable;
-} EnSyatekiManUnkStruct;
+typedef enum {
+    /* 0 */ SG_OCTO_HIT_TYPE_NONE,
+    /* 1 */ SG_OCTO_HIT_TYPE_RED,
+    /* 2 */ SG_OCTO_HIT_TYPE_BLUE,
+} ShootingGalleryoctorokHitType;
 
 typedef struct EnSyatekiMan {
-    /* 0x0000 */ Actor actor;
-    /* 0x0144 */ SkelAnime skelAnime;
-    /* 0x0188 */ EnSyatekiManActionFunc actionFunc;
-    /* 0x018C */ Path* path;
-    /* 0x0190 */ s32 unk_190;
-    /* 0x0194 */ s32 unk_194;
-    /* 0x0198 */ Vec3s jointTable[BURLY_GUY_LIMB_MAX];
-    /* 0x01F8 */ Vec3s morphTable[BURLY_GUY_LIMB_MAX];
-    /* 0x0258 */ Vec3s unk_258;
-    /* 0x025E */ Vec3s unk_25E;
-    /* 0x0264 */ s16 eyeIndex;
-    /* 0x0266 */ s16 blinkTimer;
-    /* 0x0268 */ UNK_TYPE1 unk268[0x2];
-    /* 0x026A */ s16 unk_26A;
-    /* 0x026C */ s16 unk_26C;
-    /* 0x026E */ s16 unk_26E;
-    /* 0x0270 */ s16 unk_270;
-    /* 0x0272 */ s16 unk_272;
-    /* 0x0274 */ s16 unk_274;
-    /* 0x0276 */ s16 unk_276;
-    /* 0x0278 */ s16 unk_278;
-    /* 0x027A */ s16 unk_27A;
-    /* 0x027C */ s16 unk_27C;
-    /* 0x027E */ s16 unk_27E;
-    /* 0x0280 */ s16 unk_280;
-    /* 0x0282 */ s16 unk_282;
-    /* 0x0284 */ s16 unk_284;
+    /* 0x000 */ Actor actor;
+    /* 0x144 */ SkelAnime skelAnime;
+    /* 0x188 */ EnSyatekiManActionFunc actionFunc;
+    /* 0x18C */ Path* path;
+    /* 0x190 */ s32 octorokFlags;
+    /* 0x194 */ s32 swampTargetActorListIndex; // used but never initialized, so the value is implicitly always 0
+    /* 0x198 */ Vec3s jointTable[BURLY_GUY_LIMB_MAX];
+    /* 0x1F8 */ Vec3s morphTable[BURLY_GUY_LIMB_MAX];
+    /* 0x258 */ Vec3s headRot;
+    /* 0x25E */ Vec3s torsoRot;
+    /* 0x264 */ s16 eyeIndex;
+    /* 0x266 */ s16 blinkTimer;
+    /* 0x268 */ UNK_TYPE1 unk268[0x2];
+    /* 0x26A */ s16 shootingGameState;
+    /* 0x26C */ union {
+                   s16 guaySpawnTimer;
+                   s16 octorokState;
+               } perGameVar1;
+    /* 0x26E */ union {
+                   s16 bonusDekuScrubHitCounter;
+                   s16 octorokHitType;
+               } perGameVar2;
```

Comment:
Yes, if it were me I'd have made all the game-specific stuff a union of substructs. I wouldn't be surprised if this was just two badly-named variables with and without macros originally, but of course we can't tell. If talkWaitTimer weren't there, I'd expect that it could be a union, and then `score` as the first common variable afterwards.

Reply:
Actually I wonder if this is just a work buffer, although it might not match.

Reply:
It does match, but the same thing about `bonusDekuScrubHitCounter` I said above applies here too; why would one be in a work buffer but not the other?

Reply:
I mean the entire chunk of s16s

Reply:
Ah, that makes sense. This does indeed match. But I'm still wary of it; most of these "work" variables would only be used for exactly one thing, which doesn't really "feel" like a work buffer for me. It just feels like a workaround.

Reply:
To be more specific, I don't feel like anything suggested so far actually represents what they did. I'm guessing (or really hoping) that if they did a union of substructs or a work buffer, they would've done something more efficient than what we see here.

Now, we can do whatever we want, so long as we match, but I don't think a work buffer is a good choice here. It would mislead people into thinking there is more variable reuse than there really is. I'm imagining there would be `#defines` for the various indices into the buffer like so:
```c
#define GUAY_SPAWN_TIMER 0
#define OCTOROK_STATE 0
#define BONUS_DEKU_SCRUB_HIT_COUNTER 1
```
And so on. The only way you'd notice the lack of variable reuse is by looking at this list of defines and noticing that most numbers only appear once. If you just saw `this->work[GUAY_HIT_COUNTER]++` in the code, how would you know that this specific index is literally never reused? The only way to tell would be to look somewhere else. That's why I prefer my current approach; it makes it clear which variables are reused and which aren't.

Reply:
We don't need to follow what the original devs did, if we have a cleaner way that still matches.

Reply:
That's my point though; I believe this *is* the cleanest way of doing it.

For example, making this a work buffer will mislead people into thinking that reuse is possible when it really isn't under many circumstances. If you try to change the Deku Scrub hit counter while the Swamp game is in progress, the game will simply break. In addition, a work buffer is simply uglier than using a variable. `this->dekuScrubHitCounter = 0` looks much better than `this->work[DEKU_SCRUB_HIT_COUNTER] = 0`.

As for a union of substructs, that at least makes variable accesses look solid, but I still don't know if it's cleaner. They interleaved per-game state like `bonusDekuScrubHitCounter` and `guayFlags` with variables that were intended to be shared `talkWaitTimer`. So if you made a substruct, it would *just* be the two things that are already in unions right now. Is it cleaner? Eh, I don't think so, but at least it's better than the work buffer approach.